### DISCUSSION TITLE
Front end not updating

### DISCUSSION THREAD

**Discussion by AghaShayan123** [directus/directus#16390] [Archive]
### Describe the Bug

I have updated frontend but their changes not reflecting by using command npx directus start
when I goto app folder and run npm run dev then changes are reflecting only so docker only run npx directus start command to start docker

### To Reproduce

Changes in front end and run npx direcus start

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

18.10.0

### What database are you using?

Postgress

### What browser are you using?

Chrome

### How are you deploying Directus?

locally, Docker

**Comment by br41nslug** [comment]
are you building the source before starting it in production mode?

**Reply by AghaShayan123** [reply]
how to build before starting again currently I have run npm run build in both api and app folder

**Reply by br41nslug** [reply]
be sure to use `pnpm` when running from a checkout :) to deal with the mono repo packages
https://docs.directus.io/contributing/running-locally.html#_4-install-the-dependencies-and-build-the-project

**Reply by AghaShayan123** [reply]
every time I need to run this when I have to update something in frontend?

**Reply by br41nslug** [reply]
For production yeah, when developing stick with `pnpm -F @directus/app dev`

**Reply by AghaShayan123** [reply]
for development I need to run these two commands?
pnpm -F @directus/app dev
npx directus start

**Reply by br41nslug** [reply]
That second one is for production, it's documented here: https://docs.directus.io/contributing/running-locally.html#_7-start-the-development-server

**Reply by AghaShayan123** [reply]
thanks

**Reply by AghaShayan123** [reply]
for docker where do I need to add these commands?

**Reply by br41nslug** [reply]
I'd stick with local development before production deployment via docker because rebuilding the docker image after each change can become quite time consuming.

**Reply by AghaShayan123** [reply]
in local development everything works fine we need to update docker

**Reply by br41nslug** [reply]
https://docs.docker.com/language/nodejs/build-images/
